### ISSUE TITLE
watchQuery unsubscription

### ISSUE THREAD

**Issue by levimoore** [ember-graphql/ember-apollo-client#353]
If my route is set up like this, do I need to handle unsubscribing from the watchQuery because I am extending the apollo service with my own? or does the queryManager still handle unsubscription? 

```
import Route from "@ember/routing/route";
import { inject as service } from "@ember/service";
import { queryManager } from "ember-apollo-client";
import { tracked } from "@glimmer/tracking";
import query from "../../queries/findUserByUid.graphql";

export default class LeaguesRoute extends Route {
  @service auth;
  @queryManager({ service: "fauna-apollo" }) apollo;
  @tracked uid;

  beforeModel() {
    this.uid = this.auth.currentUserUid();
  }

  model() {
    let variables = { uid: this.uid };
    return this.apollo.watchQuery({
      query,
      variables,
    });
  }
}
```
When I first enter the route my data loads fine, but if I hit refresh, watchQuery just resolves as an empty object, even though the network tab show the data in the response....

If I do need to unsubscribe. How do I?

**Comment by levimoore** [comment]
I figured out my problem based on the answer [here.](https://github.com/apollographql/react-apollo/issues/2925)

In my query I had some nested data where one object type was part of two separate sections of the query. I had returned `_id` for one but not the other. I think this results in Apollo's `InMemoryCache` creating an Id for the one that doesn't have one. This causes the cache not to resolve and return an empty object.

TLDR; Return ID's for all the things. 


Ignore the below....
ember-apollo-client returning empty object
ember watchQuery returns empty object network tab shows data
ember model hook empty object
ember route Promise empty
